In [2]:
import pandas as pd

df = pd.read_csv('/content/Messy_Employee_dataset.csv')

df.head()

,Employee_ID,First_Name,Last_Name,Age,Department_Region,Status,Join_Date,Salary,Email,Phone,Performance_Score,Remote_Work
0,EMP1000,Bob,Davis,25.0,DevOps-California,Active,4/2/2021,59767.65,bob.davis@example.com,-1651623197,Average,True
1,EMP1001,Bob,Brown,NaN,Finance-Texas,Active,7/10/2020,65304.66,bob.brown@example.com,-1898471390,Excellent,True
2,EMP1002,Alice,Jones,NaN,Admin-Nevada,Pending,12/7/2023,88145.90,alice.jones@example.com,-5596363211,Good,True
3,EMP1003,Eva,Davis,25.0,Admin-Nevada,Inactive,11/27/2021,69450.99,eva.davis@example.com,-3476490784,Good,True
4,EMP1004,Frank,Williams,25.0,Cloud Tech-Florida,Active,1/5/2022,109324.61,frank.williams@example.com,-1586734256,Poor,False


In [3]:
print("Dataset Shape:", df.shape)

df.head()

Dataset Shape: (1020, 12)


,Employee_ID,First_Name,Last_Name,Age,Department_Region,Status,Join_Date,Salary,Email,Phone,Performance_Score,Remote_Work
0,EMP1000,Bob,Davis,25.0,DevOps-California,Active,4/2/2021,59767.65,bob.davis@example.com,-1651623197,Average,True
1,EMP1001,Bob,Brown,NaN,Finance-Texas,Active,7/10/2020,65304.66,bob.brown@example.com,-1898471390,Excellent,True
2,EMP1002,Alice,Jones,NaN,Admin-Nevada,Pending,12/7/2023,88145.90,alice.jones@example.com,-5596363211,Good,True
3,EMP1003,Eva,Davis,25.0,Admin-Nevada,Inactive,11/27/2021,69450.99,eva.davis@example.com,-3476490784,Good,True
4,EMP1004,Frank,Williams,25.0,Cloud Tech-Florida,Active,1/5/2022,109324.61,frank.williams@example.com,-1586734256,Poor,False


In [4]:
# Check dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1020 entries, 0 to 1019
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Employee_ID        1020 non-null   object 
 1   First_Name         1020 non-null   object 
 2   Last_Name          1020 non-null   object 
 3   Age                809 non-null    float64
 4   Department_Region  1020 non-null   object 
 5   Status             1020 non-null   object 
 6   Join_Date          1020 non-null   object 
 7   Salary             996 non-null    float64
 8   Email              1020 non-null   object 
 9   Phone              1020 non-null   int64  
 10  Performance_Score  1020 non-null   object 
 11  Remote_Work        1020 non-null   bool   
dtypes: bool(1), float64(2), int64(1), object(8)
memory usage: 88.8+ KB


In [5]:
# Check missing values in each column

missing_values = df.isnull().sum()

print(missing_values)

Employee_ID            0
First_Name             0
Last_Name              0
Age                  211
Department_Region      0
Status                 0
Join_Date              0
Salary                24
Email                  0
Phone                  0
Performance_Score      0
Remote_Work            0
dtype: int64


In [6]:
# Check age statistics

print(df["Age"].describe())

count    809.000000
mean      32.484549
std        5.656860
min       25.000000
25%       25.000000
50%       30.000000
75%       40.000000
max       40.000000
Name: Age, dtype: float64


In [7]:
# Check age value distribution

print(df["Age"].value_counts().sort_index())

Age
25.0    206
30.0    205
35.0    188
40.0    210
Name: count, dtype: int64


In [8]:
# Fill missing Age values with the median

age_median = df["Age"].median()

df["Age"] = df["Age"].fillna(age_median)

print("Median Age used:", age_median)
print("Missing Age values after cleaning:", df["Age"].isnull().sum())

Median Age used: 30.0
Missing Age values after cleaning: 0


In [9]:
# Check salary statistics

print(df["Salary"].describe())

count       996.000000
mean      85155.056396
std       19873.727918
min       50047.320000
25%       68392.487500
50%       85547.870000
75%      100974.027500
max      119971.650000
Name: Salary, dtype: float64


In [10]:
# Fill missing Salary values with the median

salary_median = df["Salary"].median()

df["Salary"] = df["Salary"].fillna(salary_median)

print("Median Salary used:", salary_median)
print("Missing Salary values after cleaning:", df["Salary"].isnull().sum())

Median Salary used: 85547.87
Missing Salary values after cleaning: 0


In [11]:
# Final missing values check

print(df.isnull().sum())

Employee_ID          0
First_Name           0
Last_Name            0
Age                  0
Department_Region    0
Status               0
Join_Date            0
Salary               0
Email                0
Phone                0
Performance_Score    0
Remote_Work          0
dtype: int64


In [12]:
# Check for duplicate rows

duplicate_count = df.duplicated().sum()

print("Number of duplicate rows:", duplicate_count)

Number of duplicate rows: 0


In [13]:
# Check current data type of Join_Date

print("Before conversion:", df["Join_Date"].dtype)

Before conversion: object


In [14]:
# Convert Join_Date from object to datetime

df["Join_Date"] = pd.to_datetime(df["Join_Date"])

print("After conversion:", df["Join_Date"].dtype)

After conversion: datetime64[ns]


In [15]:
# Check data types after cleaning

print(df.dtypes)

Employee_ID                  object
First_Name                   object
Last_Name                    object
Age                         float64
Department_Region            object
Status                       object
Join_Date            datetime64[ns]
Salary                      float64
Email                        object
Phone                         int64
Performance_Score            object
Remote_Work                    bool
dtype: object


In [16]:
# Check phone number values

print(df["Phone"].head(10))
print("\nData type:", df["Phone"].dtype)

0   -1651623197
1   -1898471390
2   -5596363211
3   -3476490784
4   -1586734256
5   -5409003485
6   -4518376063
7   -4134327559
8   -4177656123
9   -8156985699
Name: Phone, dtype: int64

Data type: int64


In [17]:
# Check phone number range

print("Minimum Phone value:", df["Phone"].min())
print("Maximum Phone value:", df["Phone"].max())

print("\nNumber of negative phone values:",
      (df["Phone"] < 0).sum())

Minimum Phone value: -9994973202
Maximum Phone value: -3896086

Number of negative phone values: 1020


In [18]:
# Check whether negative phone values follow any pattern

print(df["Phone"].describe())

count    1.020000e+03
mean    -4.942253e+09
std      2.817326e+09
min     -9.994973e+09
25%     -7.341992e+09
50%     -4.943997e+09
75%     -2.520391e+09
max     -3.896086e+06
Name: Phone, dtype: float64


In [19]:
# Check if Phone column contains any valid positive values

valid_phone_count = (df["Phone"] > 0).sum()

print("Valid positive phone values:", valid_phone_count)
print("Invalid negative phone values:", (df["Phone"] < 0).sum())

Valid positive phone values: 0
Invalid negative phone values: 1020


In [20]:
# Remove the invalid Phone column

df = df.drop(columns=["Phone"])

print("Phone column removed.")
print("New dataset shape:", df.shape)

Phone column removed.
New dataset shape: (1020, 11)


In [21]:
# Display basic statistics for numerical columns

print(df.describe())

               Age                      Join_Date         Salary
count  1020.000000                           1020    1020.000000
mean     31.970588  2022-07-26 12:53:38.823529472   85164.299069
min      25.000000            2020-01-01 00:00:00   50047.320000
25%      30.000000            2021-04-12 06:00:00   68811.232500
50%      30.000000            2022-08-09 12:00:00   85547.870000
75%      35.000000            2023-10-31 12:00:00  100372.662500
max      40.000000            2024-12-29 00:00:00  119971.650000
std       5.136901                            NaN   19638.385740


In [22]:
# Count employees by department and region

print(df["Department_Region"].value_counts())

Department_Region
HR-Florida               41
DevOps-California        35
Sales-Nevada             35
Admin-Nevada             34
Admin-California         34
DevOps-Florida           34
Finance-Illinois         33
DevOps-New York          33
Sales-Florida            33
HR-New York              33
DevOps-Illinois          33
Finance-California       32
Sales-California         31
Sales-Illinois           30
Admin-Illinois           29
Finance-Texas            29
Cloud Tech-Texas         29
Cloud Tech-California    29
Finance-Nevada           29
Cloud Tech-Florida       28
DevOps-Nevada            27
DevOps-Texas             27
Sales-New York           26
HR-California            26
Admin-Florida            25
Cloud Tech-New York      24
Finance-Florida          24
HR-Illinois              24
HR-Nevada                24
HR-Texas                 23
Finance-New York         23
Sales-Texas              23
Admin-Texas              22
Admin-New York           22
Cloud Tech-Nevada        20
Cl

In [23]:
# Count employees by status

print(df["Status"].value_counts())

Status
Pending     356
Active      352
Inactive    312
Name: count, dtype: int64


In [24]:
# Count employees by performance score

print(df["Performance_Score"].value_counts())

Performance_Score
Good         270
Average      267
Excellent    267
Poor         216
Name: count, dtype: int64


In [25]:
# Final dataset verification

print("Final dataset shape:", df.shape)

print("\nMissing values:")
print(df.isnull().sum())

print("\nDuplicate rows:", df.duplicated().sum())

print("\nData types:")
print(df.dtypes)

Final dataset shape: (1020, 11)

Missing values:
Employee_ID          0
First_Name           0
Last_Name            0
Age                  0
Department_Region    0
Status               0
Join_Date            0
Salary               0
Email                0
Performance_Score    0
Remote_Work          0
dtype: int64

Duplicate rows: 0

Data types:
Employee_ID                  object
First_Name                   object
Last_Name                    object
Age                         float64
Department_Region            object
Status                       object
Join_Date            datetime64[ns]
Salary                      float64
Email                        object
Performance_Score            object
Remote_Work                    bool
dtype: object


In [28]:
# Save the cleaned dataset as a CSV file

df.to_csv("cleaned_employee_dataset.csv", index=False)

print("Cleaned dataset saved successfully!")

Cleaned dataset saved successfully!


Task 1: Data Cleaning & Exploration

In this task, I cleaned and explored an employee dataset using Pandas.

Handled missing values in Age and Salary using median values.
Checked and removed duplicate records; no duplicates were found.
Converted Join_Date from object to datetime format.
Identified and removed invalid phone number data.
Analyzed basic statistics such as mean, median, minimum, and maximum.
Used value_counts() to analyze employee status and performance scores.

Final dataset: 1,020 rows × 11 columns with no missing values or duplicates.